### PR TITLE
Change process for writing release notes

### PR DESCRIPTION
**Pull request by benesch**
Change the process for writing release notes so that the notes are
written in the PR description rather than added directly to
release-notes.md. This avoids constant merge conflicts on
release-notes.md.

The release process already entails a complete survey of all PRs to
migrate only the release notes for the PRs included in the release to
the correct version heading, so this doesn't add as much of a burden to
the release process as it might initially seem.

### Motivation

* This PR fixes an increasingly common complaint with the existing release notes process.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -51,4 +51,5 @@ Delete this section if no tips.
 ### Checklist
 
 - [ ] This PR has adequate test coverage / QA involvement has been duly considered.
-- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).
+- [ ] This PR includes the following [user-facing behavior changes](/doc/developer/guide-changes.md#what-changes-require-a-release-note):
+  - <!-- Add release notes here or explicitly state that there are no user-facing behavior changes. -->
```
Comment (andrioni) [inline .github/pull_request_template.md:55]: Maybe make this into its own H3 section? Sounds easier to copy and paste later.

**Review by andrioni — APPROVED** [review]
LGTM! Should we also change the release guide to include something about collecting all the release notes?

File: .github/pull_request_template.md (hunk 1 above)
Comment (benesch) [reply .github/pull_request_template.md:57]: Done.

**Comment by benesch** [comment]
> Should we also change the release guide to include something about collecting all the release notes?

It's not actually something the release manager is meant to handle! It's the responsibility of the "release notes team", which right now is just me and de facto @morsapaes when I'm out. So I think it's okay that it lives in our heads for now, though we should get it written down eventually for sure.

**Pull request merged**